app/test/integration_test.go

```diff
@@ -172,6 +173,80 @@ func (s *IntegrationTestSuite) TestMaxBlockSize() {
 	}
 }
 
+func (s *IntegrationTestSuite) TestSharesInclusionProof() {
+	require := s.Require()
+	val := s.network.Validators[0]
+
+	// generate 100 randomly sized txs (max size == 100kb)
+	txs, err := generateSignedWirePayForBlobTxs(val.ClientCtx, s.cfg.TxConfig, s.kr, -1, s.accounts...)
+	require.NoError(err)
+
+	hashes := make([]string, len(txs))
+
+	for i, tx := range txs {
+		res, err := val.ClientCtx.BroadcastTxSync(tx)
+		require.NoError(err)
+		require.Equal(abci.CodeTypeOK, res.Code)
+		hashes[i] = res.TxHash
+	}
+
+	// wait a few blocks to clear the txs
+	for i := 0; i < 50; i++ {
```

Comment:
[blocking]
50 is way too large and will significantly increase the time of the test, I think we should lower this

Reply:
worst case would be about 10-20 iirc, but we should just use fewer or smaller PFBs imo

Reply:
I did 20 and I ran it multiple times, and sometimes some  transactions don't get included... will reduce the number of PFBs then. Thanks.

Reply:
[fe0b7b2](https://github.com/celestiaorg/celestia-app/pull/1072/commits/fe0b7b2d9889a6dbe6079c07ff403c724615f80e)